src/SourceFile.js

```diff
@@ -119,91 +148,82 @@ class SourceFile {
      * The content of the file, stored as regular Javascript string
      * encoded in UTF-8.
      *
-     * @type {String}
+     * @type {String|undefined}
      * @protected
      */
     content;
-    
+
     /**
      * Get the content of this file encoded as a regular Javascript
      * string.
-     * @returns {String} the content of the file, encoded as a JS string
+     * @returns {String|undefined} the content of the file, encoded as a JS string
      */
     getContent() {
         if (this.content === undefined) {
             this.read();
         }
         return this.content;
     }
-    
+
     /**
      * Get the content of this file, encoded as an array of lines. Each
      * line has its trailing newline character removed.
-     * @returns {Array.<String>} the content as an array of lines
+     * @returns {Array.<String>|undefined} the content as an array of lines
      */
     getLines() {
-        if (this.content === undefined) {
+        if (typeof(this.content) === 'undefined') {
             this.read();
         }
-        return this.content.split(/[\r\n]+/g);
-    }
-
-    /**
-     * Set the content of this file to the given array of lines. Each
-     * line should not have a trailing newline character. The file
-     * remains modified in-memory only and is not written out to disk
-     * until the {@link "SourceFile#write"} method is called. The newlines
-     * (LF only) are re-added before the contents are written back to disk.
-     * @param {Array.<String>} lines the lines to set as the content
-     */
-    setLines(lines) {
```

Comment:
nit: technically speaking, this is a breaking API change

Reply:
Maybe I should leave this in, but deprecated. The linter should never call it any more.